lib/std/stropt.nim

```diff
@@ -1,36 +1,38 @@
-import std/strutils
-import std/private/since
-
-
-proc stripInplace*(s: var string, leading = true, trailing = true,
-                   chars: set[char] = Whitespace) {.since: (1, 5, 1).} =
+#
+#
+#           The Nim Compiler
+#        (c) Copyright 2020 Nim Contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+proc strip*(s: var string, leading = true, trailing = true,
```

Comment:
bug:
```nim
when true:
  import std/stropt
  proc main()=
    var a = "xxxxxx"
    a.strip(chars={'x'})
  static: main()
  main()
```

CT: Error: illegal conversion from '-6' to '[0..9223372036854775807]'
RT: Error: unhandled exception: index 6 not in 0 .. 5 [IndexDefect]

Reply:
Yeah, it needs:
```nim
  if first >= last:
    setLen(s, 0)
    return
```

Reply:
but why not the way i wrote it?
```nim
if leading:
    while first < last and s[first] in chars: inc(first)
  if trailing:
    while last > first and s[last] in chars: dec(last)
```
which avoids going through an item more than once

Reply:
EDIT: i was off by one, should've been:
```nim
  var
    first = 0
    last = high(s)
  if leading:
    while first <= last and s[first] in chars: inc(first)
  if trailing:
    while last > first and s[last] in chars: dec(last)
```